client/components/theme-upgrade-modal/index.tsx

```diff
@@ -289,42 +354,24 @@ export const ThemeUpgradeModal = ( {
 	};
 
 	const getExternallyManagedPurchaseModalData = (): UpgradeModalContent => {
-		const getBusinessPlanByTerm = ( term: string ) => {
+		const getMarketplacePlanTextByTerm = ( term: string, cost: string ) => {
 			switch ( term ) {
-				case TERM_TRIENNIALLY:
-					return triAnnualBusinessPlanProduct;
-				case TERM_BIENNIALLY:
-					return biAnnualBusinessPlanProduct;
-				case TERM_ANNUALLY:
-					return businessPlanProduct;
-				case TERM_MONTHLY:
-				default:
-					return monthlyBusinessPlanProduct;
-			}
-		};
-
-		const getPlanTextByTerm = ( term: string, cost: string ) => {
-			switch ( term ) {
-				case TERM_TRIENNIALLY:
+				case 'three years':
 					return translate( '%(cost)s per three years', { args: { cost } } );
-				case TERM_BIENNIALLY:
+				case 'two years':
```

Comment:
Previously term was derived from the current plan by UnifiedDesignPicker and then passed to this component as a prop. This component then switched on that term and the implicit required plan to figure out which product was the right one.

In this PR I've changed so that the UnifiedDesignPicker figures out the product (as it was already doing it) and just uses the product as the prop, with this component getting the term from the product. It seemed redundant and more error prone to pass in both a product prop and then a term prop

However the [product field](https://github.com/Automattic/wp-calypso//blob/b5eef1489c88b45d8c0bca6cfcbd44554675e746/packages/data-stores/src/products-list/queries/use-products.ts#L42) (`product_term` from the /products API) uses totally different values to the [plans objects](https://github.com/Automattic/wp-calypso//blob/bce4bcfe6e6a29b6fca7a808cf3fe7f942f73c7e/packages/calypso-products/src/plans-list.tsx#L2355).

Reply:
I'm open to going back to passing in a separate term prop derived from the current plan, and we can then compare that against the consts as before.

Reply:
No worries, I was just curious about it.

Reply:
it irritated me while I was implementing tbh, i'm sure there's a good reason for them using different values though 😬